src/network_models/network_constructor.jl

```diff
@@ -46,21 +46,53 @@ end
 function construct_network!(
     psi_container::PSIContainer,
     sys::PSY.System,
-    ::Type{T},
+    ::Type{T};
+    instantiate_model = instantiate_nip_expr_model,
 ) where {T <: PM.AbstractPowerModel}
-    incompat_list = [
-        PM.SDPWRMPowerModel,
-        PM.SparseSDPWRMPowerModel,
-        PM.SOCBFPowerModel,
-        PM.SOCBFConicPowerModel,
-    ]
-    if T in incompat_list
+    if T in UNSUPPORTED_POWERMODELS
         throw(ArgumentError("$(T) formulation is not currently supported in PowerSimulations"))
     end
 
     get_slack_variables(psi_container.settings) && add_slacks!(psi_container, T)
 
-    powermodels_network!(psi_container, T, sys)
+    @debug "Building the $T network with $instantiate_model method"
+    powermodels_network!(psi_container, T, sys, instantiate_model)
     add_pm_var_refs!(psi_container, T, sys)
     return
 end
+
+function construct_network!(
+    psi_container::PSIContainer,
+    sys::PSY.System,
+    ::Type{T};
+    instantiate_model = instantiate_bfp_expr_model,
+) where {T <: PM.AbstractBFModel}
+    if T in UNSUPPORTED_POWERMODELS
+        throw(ArgumentError("$(T) formulation is not currently supported in PowerSimulations"))
+    end
+
+    get_slack_variables(psi_container.settings) && add_slacks!(psi_container, T)
+
+    @debug "Building the $T network with $instantiate_model method"
+    powermodels_network!(psi_container, T, sys, instantiate_model)
+    add_pm_var_refs!(psi_container, T, sys)
+    return
+end
+
+function construct_network!(
+    psi_container::PSIContainer,
+    sys::PSY.System,
+    ::Type{T};
+    instantiate_model = instantiate_vip_expr_model,
+) where {T <: PM.AbstractIVRModel}
+    if T in UNSUPPORTED_POWERMODELS
+        throw(ArgumentError("$(T) formulation is not currently supported in PowerSimulations"))
+    end
+
+    get_slack_variables(psi_container.settings) && add_slacks!(psi_container, T)
+
+    @debug "Building the $T network with $instantiate_model method"
+    powermodels_network!(psi_container, T, sys, instantiate_model)
+    #add_pm_var_refs!(psi_container, T, sys) #TODO: fix this for quadratic variables
```

Comment:
Is this still a TODO: ?

Reply:
kinda, but It's bigger than just the variable references. It's for the entire build VI power models code path. The initial hooks for that code path are in here but commented/nonfunctional at the moment. I never know what to do in these cases. Should I remove the commented code, or leave it commented out? @daniel-thom might have opinions.